components/host-dashboard/reports-section/TransactionsOverviewSection.js

```diff
@@ -0,0 +1,258 @@
+      }
+      expenseStats(accounts: $accounts, from: $dateFrom, to: $dateTo) {
+        numExpenses
+        dailyAverage
+        numInvoices
+        numReimbursements
+        numGrants
+      }
+    }
+  }
+`;
+
+const TransactionsOverviewSection = ({ hostSlug, currency }) => {
+  const [dateFrom, setDateFrom] = useState(null);
+  const [dateTo, setDateTo] = useState(null);
+  const [collectives, setCollectives] = useState(null);
+
+  const { data, loading, refetch } = useQuery(transactionsOverviewQuery, {
+    variables: { hostSlug, dateFrom, dateTo, accounts: collectives },
+    context: API_V2_CONTEXT,
+  });
+  const contributionStats = data?.host.contributionStats;
+  const expenseStats = data?.host.expenseStats;
+
+  useEffect(() => {
+    refetch();
+  }, [dateFrom, dateTo, collectives]);
+
+  const { numContributions, numRecurring, numOneTime, dailyAvgIncome } = contributionStats || 0;
+  const { numExpenses, numInvoices, numReimbursements, numGrants, dailyAverage } = expenseStats || 0;
+
+  const setDate = dateRange => {
+    if (!dateRange) {
+      setDateFrom(null);
+      setDateTo(null);
+      refetch();
+      return;
+    }
+    const dates = dateRange.split('→');
+    const dateFrom = dates[0];
+    const dateTo = dates[1];
+    if (dateFrom === 'all') {
+      setDateFrom(null);
+    } else {
+      setDateFrom(dateFrom);
+    }
+    if (dateTo === 'all') {
+      setDateTo(null);
+    } else {
+      setDateTo(dateTo);
+    }
+  };
+
+  const getDateString = () => {
+    let dateString;
+    if (!dateFrom && !dateTo) {
+      return;
+    }
+    if (dateFrom) {
+      dateString = `${dateFrom}→`;
+    } else {
+      dateString = 'all→';
+    }
+
+    if (dateTo) {
+      dateString += dateTo;
+    } else {
+      dateString += 'all';
+    }
+    return dateString;
+  };
+
+  const setCollectiveFilter = collectives => {
+    if (collectives.length === 0) {
+      setCollectives(null);
+    } else {
+      const collectiveIds = collectives.map(collective => ({ legacyId: collective.value.id }));
+      setCollectives(collectiveIds);
+    }
+  };
+
+  return (
+    <React.Fragment>
+      <Flex flexWrap="wrap">
+        <Container width={[1, 1, 1 / 2]} pr={2} mb={[3, 3, 0, 0]}>
+          <FilterLabel htmlFor="transactions-period-filter">
+            <FormattedMessage id="TransactionsOverviewSection.PeriodFilter" defaultMessage="Filter by Date" />
+          </FilterLabel>
+          <PeriodFilter onChange={value => setDate(value)} value={getDateString()} />
+        </Container>
+        <Container width={[1, 1, 1 / 2]}>
+          <FilterLabel htmlFor="transactions-collective-filter">
+            <FormattedMessage id="TransactionsOverviewSection.CollectiveFilter" defaultMessage="Filter by Collective" />
+          </FilterLabel>
+          <CollectivePickerAsync
+            inputId="TransactionsCollectiveFilter"
+            data-cy="transactions-collective-filter"
+            types={['COLLECTIVE']}
```

Comment:
I wonder whether we want to ignore projects and events here. What do you think?

Reply:
My thinking is yes to ignore projects and events for at least the MVP. 

In the design all the sections have "Filter by Collective" and the words "Collective(s)" are used throughout giving the user the impression this reports section is about collectives not events or projects. Thus I think we have to only show collectives unless we make it explicit to the user somewhere in the screen that they are seeing projects and events as well. That's my take on it. 😉

Reply:
But what would be the host's expectations when they visit this page? To me, it sounds like they would want to see all transactions happening under their fiscal hosting, regardless of the account types. The value proposition of this section is to show a clear view of how the money is flowing among your hosted initiatives; yet if you limit that to collectives you'll miss a great part of the transactions (some collectives now manage their budgets mainly through projects).

I currently don't see any good reason to limit that to collectives. If it's a wording problem, then we should update the wording and say "Accounts" or "Hosted initiatives" rather than "Collectives"/

It would be nice to merge this PR, so let's not block on this but we should ask for design and iterate soon enough.

Reply:
Thanks for the information. I've created an issue for this at, https://github.com/opencollective/opencollective/issues/4726 👍🏽

Reply:
@Betree : As per our discussion, I've added the support for filtering by projects and events at 59d3a87.